benchmark/Streamly/Benchmark/Data/Array/Foreign/Mut.hs

```diff
@@ -52,21 +53,25 @@ benchIO name src sink =
 
 o_1_space_serial_marray :: Int -> MArray.Array Int -> [Benchmark]
 o_1_space_serial_marray value array =
-    [ benchIO "partitionBy (< 0)" (const array)
-        $ MArray.partitionBy (< 0)
-    , benchIO "partitionBy (> 0)" (const array)
-        $ MArray.partitionBy (> 0)
-    , benchIO "partitionBy (< value/2)" (const array)
-        $ MArray.partitionBy (< (value `div` 2))
-    , benchIO "partitionBy (> value/2)" (const array)
-        $ MArray.partitionBy (> (value `div` 2))
-    , benchIO "strip (< value/2 || > value/2)" (const array)
-        $ MArray.strip (\x -> x < value `div` 2 || x > value `div` 2)
-    , benchIO "strip (> 0)" (const array)
-        $ MArray.strip (> 0)
-    , benchIO "modifyIndices id" (const array)
-        $ Stream.fold (MArray.modifyIndices id array) . (IS.fromStreamD . MArray.toStreamD)
-    ]
+    let idxarr = unsafePerformIO
+                $ MArray.fromStream (sourceUnfoldrM value 0 :: SerialT IO Int)
+    in [ benchIO "partitionBy (< 0)" (const array)
+            $ MArray.partitionBy (< 0)
+        , benchIO "partitionBy (> 0)" (const array)
+            $ MArray.partitionBy (> 0)
+        , benchIO "partitionBy (< value/2)" (const array)
+            $ MArray.partitionBy (< (value `div` 2))
+        , benchIO "partitionBy (> value/2)" (const array)
+            $ MArray.partitionBy (> (value `div` 2))
+        , benchIO "strip (< value/2 || > value/2)" (const array)
+            $ MArray.strip (\x -> x < value `div` 2 || x > value `div` 2)
+        , benchIO "strip (> 0)" (const array)
+            $ MArray.strip (> 0)
+        , benchIO "modifyIndices (+ 2)" (const idxarr)
+            $ Stream.fold
+            (MArray.modifyIndices (+ 2) array)
+            . (Stream.fromStreamD . MArray.toStreamD)
```

Comment:
We should not use StreamD functions. StreamD is a purely internal module. Isn't there a `toStream` function in MArray?